realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -2216,6 +2217,56 @@ private OsResults lazyFindAll() {
         return this;
     }
 
+    /**
+     * Create a text-based predicate using the Realm Query Language. This predicate can be combined
+     * with other raw or type safe predicates.
+     * <p>
+     * Class and property names used in the raw predicate can be either the names defined in the
+     * Realm Model classes or the internal names defined using the {@link io.realm.annotations.RealmClass}
+     * or {@link io.realm.annotations.RealmField} annotations.
+     * </p>
+     * The string-based predicates can contain placeholders that conform to the syntax supported by
+     * {@link java.util.Formatter}.
+     * <p>
+     * See <a href="https://docs.mongodb.com/realm-sdks/js/latest/tutorial-query-language.html">these docs</a>
+     * for a more detailed description of the Realm Query Language.
+     * <p>
+     * <pre>
+     * Examples:
+     * {@code
+     * RealmQuery<Person> query = realm.where(Person.class);
+     *
+     * // Simple query
+     * query.rawPredicate("name = 'Jane'");
+     * query.rawPredicate("name = '%s'", getName());
```

Comment:
You will find syntax errors if the string being substituted here contains any single quote character without escaping it. The recommended way is to use the query parser's arguments (`name = $1`) and pass the string variable in to the query parser in an argument list, then you don't have to do any special character escaping. The `Table::query` method accepts a `vector<Mixed>` for an argument list for this purpose. Not blocking though if you want to get this in for now.

Reply:
Hmm, could point. I used the formatting from Java because it was easier instead of trying to marshell parameters across JNI, but that was probably a bad idea. By using the parser substitution it will also make it easier to document.

Reply:
It turns out that supporting the Mixed argument list isn't trivial yet until we have the full Mixed support currently being added. Instead of spending time adding work-arounds, I'm removing support for argument substitution until we have full Mixed support. This shouldn't take long anyway and the work-arounds are pretty clear and simple, especially in Kotlin with already support in-place substitution of variables.